src/server/session.ts

```diff
@@ -573,7 +586,7 @@ namespace ts.server {
 
         // In the common case where there's only one project, return a simpler result to make
         // it easier for the caller to skip post-processing.
-        if (searchedProjects.size === 1) {
+        if (resultsMap.size === 1) {
             const it = resultsMap.values().next();
             return it.done ? emptyArray : it.value; // There may not be any results at all

```

Comment:
Now that you actually are checking the `resultsMap` instead of `searchedProjects`, is there ever a case where `it.done` doesn't imply a value?

Reply:
You're half-right: now that `resultsMap` contain undefined values, that might be undefined.  I think I caught that downstream, but this would be better.

Reply:
Yes, _but_ `it.value` can now be null.  Updating.

Reply:
No, it can't - I already prevented that by construction.